doc/man3/EVP_PKEY_sign.pod

```diff
@@ -26,37 +35,121 @@ for more information about implicit fetches.
+
+=end comment
+
+=head2 General
+
+Some signature implementations only accumulate the input data and do no
+further processing before signing it (they expect the input to be a digest),
+while others compress the data, typically by internally producing a digest,
+and signing the result.
+Some of them support both modes of operation at the same time.
+The caller is expected to know how the chosen algorithm is supposed to behave
+and under what conditions.
+
+For example, an RSA implementation can be expected to only expect a digest as
+input, while ED25519 can be expected to process the input with a hash, i.e.
+to produce the digest internally, and while RSA-SHA256 can be expected to
+handle either mode of operation, depending on if the operation was initialized
+with EVP_PKEY_sign_init_ex2() or with EVP_PKEY_sign_message_init().
+
+Similarly, an RSA implementation usually expects additional details to be set,
+like the message digest algorithm that the input is supposed to be digested
+with, as well as the padding mode (see L<EVP_PKEY_CTX_set_signature_md(3)> and
+L<EVP_PKEY_CTX_set_rsa_padding(3)> and similar others), while an RSA-SHA256
+implementation usually has these details pre-set and immutable.
+
+The functions described here can't be used to combine separate algorithms.  In
+particular, neither L<EVP_PKEY_CTX_set_signature_md(3)> nor the B<OSSL_PARAM>
+parameter "digest" (B<OSSL_SIGNATURE_PARAM_DIGEST>) can not be used to combine
+a signature algorithm with a hash algorithm to process the input.  In other
+words, it's not possible to specify a I<ctx> pre-loaded with an RSA pkey, or
+an I<algo> that fetched C<RSA> and try to specify SHA256 separately to get the
+functionality of RSA-SHA256.  If combining algorithms in that manner is
+desired, please use L<EVP_DigestSignInit(3)> and associated functions, or
+L<EVP_SignInit(3)> and associated functions.
+
+=head2 Performing multiple signatures
+
+EVP_PKEY_sign_message_final() resets the context I<ctx> in such a way that
+another sequence of (multiple) EVP_PKEY_sign_message_update() and
+EVP_PKEY_sign_message_final() can be performed using the parameters that was
+given by the initializing call.
+Likewise, EVP_PKEY_sign() can be called more than once on the same context to
+have several one-shot operations are performed using the same parameters.
 
 =head1 RETURN VALUES
 
-EVP_PKEY_sign_init() and EVP_PKEY_sign() return 1 for success and 0
-or a negative value for failure. In particular a return value of -2
-indicates the operation is not supported by the public key algorithm.
+All functions return return 1 for success and 0 or a negative value for failure.
+
+In particular, EVP_PKEY_sign_init() and its other variants may return -2 to
+indicate that the operation is not supported by the public key algorithm.
 
 =head1 EXAMPLES
 
-Sign data using RSA with PKCS#1 padding and SHA256 digest:
+=begin comment
+
+These examples are largely replicated in EVP_PKEY_sign.pod, please keep them
```

Comment:
?? This file is EVP_PKEY_sign

Reply:
Ah, thanks.  This should say `EVP_PKEY_verify.pod`, of course...